var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -200,53 +183,22 @@ def setup_build_environment(self, env):
         if self.spec.satisfies("@:1.8"):
             self.spec["py-numpy"].package.setup_build_environment(env)
 
-    # TODO: remove once pip build supports BLAS/LAPACK specification
-    # https://github.com/mesonbuild/meson-python/pull/167
     @when("@1.9:")
-    def install(self, spec, prefix):
-        blas = spec["blas"].libs.names[0]
-        lapack = spec["lapack"].libs.names[0]
-        if spec["blas"].name in ["intel-mkl", "intel-parallel-studio", "intel-oneapi-mkl"]:
-            blas = "mkl-dynamic-lp64-seq"
-        if spec["lapack"].name in ["intel-mkl", "intel-parallel-studio", "intel-oneapi-mkl"]:
-            lapack = "mkl-dynamic-lp64-seq"
-        if spec["blas"].name in ["blis", "amdblis"]:
-            blas = "blis"
-        if "armpl" in blas:
-            if "_mp" in blas:
-                blas = "armpl-dynamic-lp64-omp"
-            else:
-                blas = "armpl-dynamic-lp64-seq"
-        if "armpl" in lapack:
-            if "_mp" in lapack:
-                lapack = "armpl-dynamic-lp64-omp"
-            else:
-                lapack = "armpl-dynamic-lp64-seq"
-
-        args = [
-            "setup",
-            "build",
-            "-Dblas=" + blas,
-            "-Dlapack=" + lapack,
-            "--prefix=" + join_path(os.getcwd(), "build-install"),
-            "-Ddebug=false",
-            "-Doptimization=2",
-        ]
-        meson = which("meson")
-        meson(*args)
-        args = [
-            "-m",
-            "build",
-            "--wheel",
-            "-Cbuilddir=build",
-            "--no-isolation",
-            "--skip-dependency-check",
-            "-Ccompile-args=-j%s" % make_jobs,
-            ".",
-        ]
-        python(*args)
-        args = std_pip_args + ["--prefix=" + prefix, glob.glob(join_path("dist", "scipy*.whl"))[0]]
-        pip(*args)
+    def config_settings(self, spec, prefix):
+        blas, lapack = self.spec["py-numpy"].package.blas_lapack_pkg_config()
+        return {
+            "compile-args": f"-j{make_jobs}",
+            # http://scipy.github.io/devdocs/building/blas_lapack.html
+            "setup-args=-Dblas": blas,
+            "setup-args=-Dlapack": lapack,
+            "setup-args=-Ddebug": False,
+            "setup-args=-Doptimization": 2,
```

Comment:
Let me know if there are any other flags we should pass

Reply:
You shouldn't need `-Ddebug` or `-Doptimization`, unless you're trying to always override the meson-python default of `-Dbuildtype=release`, which implies an optimization level of 3.

What does `blas_lapack_pkg_config` return, a string name? And for MKL the relevant pkg-config string?

Reply:
Yes, it returns the pkg-config string names for both blas and lapack. It's the same code we had as before, just moved to numpy so it isn't duplicated

Reply:
Thanks - could have seen that from a full review. This all LGTM, and no other flags need passing.